### PR TITLE
Added support for POS embedding

### PR DESCRIPTION
**Pull request by yechenh**

### CODE REVIEW

File: lasertagger/custom_utils.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import json
+import os
+import nltk
+
+from typing import Text, List
+nltk.download('averaged_perceptron_tagger')
+
+
+        
+def convert_to_POS(tokens: List[Text]) -> List[int]:
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:26]: Usually the name of a function should be consistent in use of cases.
Either
convert_to_pos
or
ConvertToPos

see the style guide: https://google.github.io/styleguide/pyguide.html#316-naming

File: lasertagger/custom_utils.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    """Convert a list of tokens to a list of int representing Part of Speech (POS) tag.
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:27]: s/int/integers

File: lasertagger/custom_utils.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    
+    Returns:
+        tokens_POS: a list of int representing the POS.
+    """
+    POS_tags = ["CC", "CD", "DT", "EX","FW", "IN", "JJ", "JJR", "JJS", "LS",\
+                "MD", "NN", "NNS", "NNP", "NNPS", "PDT", "POS", "PRP", "PRP$",\
+                "RB", "RBR", "RBS", "RP", "TO", "UH", "VB", "VBD", "VBG",\
+                "VBN", "VBP", "VBZ", "WDT", "WP", "WP$", "WRB", '.', 'X']
+    POS_dict = {}
+    count = 3 # saving 0, 1, 2 for other taggings in BERT
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:39]: s/count/pos

File: lasertagger/custom_utils.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    for i in POS_tags:
+        POS_dict[i] = count
+        count += 1
+    unidentified_tag = count
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:43]: s/unidentified_tag/oov_tag

File: lasertagger/custom_utils.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    Returns:
+        tokens_POS: a list of int representing the POS.
+    """
+    POS_tags = ["CC", "CD", "DT", "EX","FW", "IN", "JJ", "JJR", "JJS", "LS",\
+                "MD", "NN", "NNS", "NNP", "NNPS", "PDT", "POS", "PRP", "PRP$",\
+                "RB", "RBR", "RBS", "RP", "TO", "UH", "VB", "VBD", "VBG",\
+                "VBN", "VBP", "VBZ", "WDT", "WP", "WP$", "WRB", '.', 'X']
+    POS_dict = {}
+    count = 3 # saving 0, 1, 2 for other taggings in BERT
+    for i in POS_tags:
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:40]: s/i/tag

File: lasertagger/custom_utils.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        
+def convert_to_POS(tokens: List[Text]) -> List[int]:
+    """Convert a list of tokens to a list of int representing Part of Speech (POS) tag.
+    Args:
+        tokens: a list of tokens to be converted.
+    
+    Returns:
+        tokens_POS: a list of int representing the POS.
+    """
+    POS_tags = ["CC", "CD", "DT", "EX","FW", "IN", "JJ", "JJR", "JJS", "LS",\
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:34]: s/POS_tags/pos_tags

File: lasertagger/custom_utils.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    POS_dict = {}
+    count = 3 # saving 0, 1, 2 for other taggings in BERT
+    for i in POS_tags:
+        POS_dict[i] = count
+        count += 1
+    unidentified_tag = count
+    
+    toens_POS_tags = nltk.pos_tag(tokens)
+    tokens_POS = []
+    for row in toens_POS_tags:
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:47]: s/toens_POS_tags/token_pos_tags

File: lasertagger/custom_utils.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+                "RB", "RBR", "RBS", "RP", "TO", "UH", "VB", "VBD", "VBG",\
+                "VBN", "VBP", "VBZ", "WDT", "WP", "WP$", "WRB", '.', 'X']
+    POS_dict = {}
+    count = 3 # saving 0, 1, 2 for other taggings in BERT
+    for i in POS_tags:
+        POS_dict[i] = count
+        count += 1
+    unidentified_tag = count
+    
+    toens_POS_tags = nltk.pos_tag(tokens)
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:45]: s/toens_POS_tags/token_pos_tags

File: lasertagger/custom_utils.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    tokens_POS = []
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:46]: s/tokens_POS/tokens_pos

File: lasertagger/custom_utils.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+# limitations under the License
+
+"""Customized utils for modified LaserTagger model."""
+
+import json
+import os
+import nltk
+
+from typing import Text, List
+nltk.download('averaged_perceptron_tagger')
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils.py:22]: I remember you can do this in a __init__.py file under the same dir.

I have never seen any other code inside Google directly calling a function in the global scope.

File: lasertagger/custom_utils_test.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+import unittest
+from unittest import TestCase
+import custom_utils
+    
+    
+class CustomUtilsTestCase(TestCase):
+    def test_convert_to_POS(self):
+        result = custom_utils.convert_to_POS(["This is a test string"])
+        self.assertTrue(max(result) <= 40 and min(result) >= 3)
```
Comment (XiangZeng2020) [inline lasertagger/custom_utils_test.py:25]: need to check the returned result

File: lasertagger/custom_utils.py (hunk 3 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 4 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 5 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 7 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 8 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 9 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 10 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils_test.py (hunk 11 above)
Comment (yechenh) [reply lasertagger/custom_utils_test.py]: Done

File: lasertagger/custom_utils.py (hunk 1 above)
Comment (yechenh) [reply lasertagger/custom_utils.py]: Done

File: lasertagger/custom_utils.py (hunk 2 above)
Comment (yechenh) [reply lasertagger/custom_utils.py:23]: Since this is part of the typing, I think it has to be int

File: lasertagger/bert_example.py [hunk 12]
```diff
@@ -163,19 +171,28 @@ def build_bert_example(
...
     labels_mask = [0] + [1] * len(labels) + [0]
     labels = [0] + labels + [0]
 
     input_ids = self._tokenizer.convert_tokens_to_ids(input_tokens)
     input_mask = [1] * len(input_ids)
-    segment_ids = [0] * len(input_ids)
+    if self._embedding_type == "Normal":
+        segment_ids = [0] * len(input_ids)
+    elif self._embedding_type == "POS":
+        segment_ids = [2] + special_tags + [41]
```
Comment (liangji-google) [inline lasertagger/bert_example.py:191]: this looks like a magic number. define this number as const to make it clear.

**Pull request merged**